File: module_01/lab_server.py
"""
XSS Mastery Lab — Module 01
Flask lab server: minimal home page only.
Vulnerable endpoints are added in later modules — this module is about
getting the environment running and understanding what we are setting up.

Run:
    pip install -r requirements.txt
    python lab_server.py

Then open: http://localhost:5000
"""

from flask import Flask, render_template_string, send_from_directory
import os

app = Flask(__name__, static_folder="public")

# ─── Home ────────────────────────────────────────────────────────────────────

@app.route("/")
def home():
    """
    Serves the lab home page from public/index.html.
    In later modules we will add routes that are deliberately vulnerable.
    For now this just confirms the server is alive.
    """
    return send_from_directory("public", "index.html")


# ─── Health check — useful for scripting later ───────────────────────────────

@app.route("/ping")
def ping():
    return {"status": "ok", "module": 1}


# ─── Entry point ─────────────────────────────────────────────────────────────

if __name__ == "__main__":
    port = int(os.environ.get("PORT", 5000))
    print(f"\n  XSS Mastery Lab — Module 01")
    print(f"  Listening on http://localhost:{port}")
    print(f"  Press Ctrl+C to stop\n")
    # debug=True gives auto-reload and detailed error pages.
    # NEVER use debug=True in production — error pages leak internals.
    app.run(host="0.0.0.0", port=port, debug=True)

File: module_01/lab_server.js
/**
 * XSS Mastery Lab — Module 01
 * Node.js / Express lab server: minimal home page only.
 * Mirrors lab_server.py exactly — choose whichever runtime you prefer.
 * Both servers stay in sync across every module.
 *
 * Run:
 *   npm install
 *   node lab_server.js        (or: npm start)
 *   npm run dev               (auto-reload with nodemon)
 *
 * Then open: http://localhost:5000
 */

const express = require("express");
const path    = require("path");

const app  = express();
const PORT = process.env.PORT || 5000;

// ── Serve everything inside /public as static files ─────────────────────────
// express.static is the Node equivalent of Flask's send_from_directory.
// In later modules we will add route handlers BEFORE this line so that
// dynamic vulnerable routes take priority over static files.
app.use(express.static(path.join(__dirname, "public")));

// Parse URL-encoded form bodies (needed from Module 04 onward)
app.use(express.urlencoded({ extended: false }));

// Parse JSON bodies (needed from Module 09 onward)
app.use(express.json());

// ── Home ─────────────────────────────────────────────────────────────────────
// express.static already handles GET / → public/index.html automatically,
// but we add an explicit route so later modules can override it cleanly.
app.get("/", (req, res) => {
  res.sendFile(path.join(__dirname, "public", "index.html"));
});

// ── Health check ─────────────────────────────────────────────────────────────
app.get("/ping", (req, res) => {
  res.json({ status: "ok", module: 1 });
});

// ── Start ─────────────────────────────────────────────────────────────────────
app.listen(PORT, () => {
  console.log(`\n  XSS Mastery Lab — Module 01`);
  console.log(`  Listening on http://localhost:${PORT}`);
  console.log(`  Press Ctrl+C to stop\n`);
});
